server/src/main/java/com/broadcom/lsp/cobol/core/annotation/HandleShutdownState.java

```diff
@@ -0,0 +1,33 @@
+package com.broadcom.lsp.cobol.core.annotation;
+
+import com.broadcom.lsp.cobol.service.DisposableLanguageServer;
+import com.google.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+public class HandleShutdownState implements MethodInterceptor {
+  private static final String SHUTDOWN_RESPONSE = "InvalidRequest";
+  private LanguageServer server;
+
+  @Inject
+  public void setServer(LanguageServer server) {
+    this.server = server;
+  }
+
+  @Override
+  public Object invoke(MethodInvocation invocation) throws Throwable {
+    if (server instanceof DisposableLanguageServer) {
+      DisposableLanguageServer languageServer = (DisposableLanguageServer) server;
+      if (languageServer.getExitCode() == DisposableLanguageServer.SHUTDOWN_EXIT_CODE) {
+        LOG.info(invocation.getMethod().getName() + " invoked after shutdown");
+        return CompletableFuture.supplyAsync(() -> SHUTDOWN_RESPONSE);
```

Comment:
You don't really need to resolve this future asynchronously. Please, use this instead:
```
return CompletableFuture.completedFuture(SHUTDOWN_RESPONSE);
```

Reply:
done